examples/docker-compose-auth-host-traefik2.yml

```diff
@@ -0,0 +1,63 @@
+version: '3'
+
+services:
+  traefik:
+    image: traefik:2.1
+    ports:
+      - "8085:80"
+      - "8086:8080"
+      - "8087:443"
+    networks:
+      - traefik
+      - traefik-docker
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+    command:
+        - --api.dashboard=true
+        - --api.insecure=false
+        - --entrypoints.web.address=:80
+        - --entrypoints.web-secure.address=:443
+        - --providers.docker.exposedbydefault=false
+        - --log.level=DEBUG
+
+  whoami:
+    image: containous/whoami:latest
+    networks:
+      - traefik
+    labels:
+      traefik.enable: 'true'
+      traefik.http.routers.whoami.entrypoints: web
+      traefik.http.routers.whoami.middlewares: tfa-verify
+      traefik.http.routers.whoami.rule: "Host(`whoami.example.com`)"
+      traefik.http.services.whoami.loadbalancer.server.port: '80'
+
+  traefik-forward-auth:
+    image: thomseddon/traefik-forward-auth:2
+    labels:
+      traefik.enable: 'true'
+      traefik.docker.network: oauth-web
+      traefik.http.middlewares.tfa-verify.forwardAuth.address: http://traefik-forward-auth:4181
+      traefik.http.middlewares.tfa-verify.forwardAuth.trustForwardHeader: 'true'
+      traefik.http.middlewares.tfa-verify.forwardAuth.authResponseHeaders: X-Forwarded-User
+      traefik.http.routers.tfa.entrypoints: web
+      traefik.http.routers.tfa.middlewares: tfa-verify
```

Comment:
Just as a note mainly to myself: This line is very very important or you will end up in a redirect loop.

Reply:
I'm having the redirect loop as well, so I'm trying to adapt your solution locally. Doesn't seem to work in my circumstances yet.

Would you help me understand why traefik-forward-auth needs to forward auth to itself?

Could you elaborate where the `oauth-web` network is defined?

Would you help me understand if/how multiple networks are being used by traefik-forward-auth? In my case, I only have the public traefik-proxy-network that all my containers share.

Does the `hostname` need to be defined as well for the `forwardAuth.address` to work? (In Docker Swarm, hostnames are randomized.)

Reply:
Traefik needs this for whatever reason.
Traefik-forward-auth does not care about multiple networks. Traefik and network connection do care.
in swam they still have the predictable name of the service.